utils/robotouille_input.py

```diff
@@ -20,37 +28,46 @@ def create_action_from_control(env, obs, action, renderer):
     Returns:
         A valid action.
     """
-    if len(action) == 0: return
-    valid_actions = list(env.action_space.all_ground_literals(obs))
-    str_valid_actions = list(map(str, valid_actions))
+    if len(action) == 0: return None, None
+    valid_actions = (obs.get_valid_actions())
     action = action[0]
+    for predicate, value in obs.predicates.items():
+        if predicate.name == "loc" and value:
+            player_loc = str(predicate.params[1])
     if action.type == pygame.MOUSEBUTTONDOWN:
         pos_x, pos_y = action.pos
         grid_size = renderer.canvas.pix_square_size[0]
         layout_pos = int(pos_x / grid_size), int(pos_y / grid_size)
         clicked_station = renderer.canvas.layout[layout_pos[1]][layout_pos[0]]
-        loc_tuples = [str_valid_action.split(",") for str_valid_action in str_valid_actions]
-        locs = [loc_tuple[2].split(":")[0] if 'station' in loc_tuple[2] else loc_tuple[3].split(":")[0] for loc_tuple in loc_tuples]
-        if clicked_station in locs:
-            index = locs.index(clicked_station)
-            while 'cook(' in str_valid_actions[index] or 'cut(' in str_valid_actions[index] or 'fry(' in str_valid_actions[index] or 'fry_cut_item(' in str_valid_actions[index]:
-                # We look for fry( instead which should be guaranteed to only match the fry action, and not the fryer
-                index = locs.index(clicked_station, index+1)
-            return str_valid_actions[index]
+        for args in valid_actions[ACTIONS[0]]:
```

Comment:
This file is very hard to understand because of indexing into `ACTIONS`. I see you're using this as the action space as well; can you create a dictionary that maps strings to the actions so it's easier to understand what is going on, e.g. `for args in valid_actions[ACTIONS_DICT['move']]`